app/models/atmosphere/compute_site.rb

```diff
@@ -80,6 +80,8 @@ class ComputeSite < ActiveRecord::Base
               presence: true,
               inclusion: %w(openstack aws rackspace google_compute)
 
+    validate :nic_provider_class_defined, unless: "nic_provider_class_name.nil?"
```

Comment:
Maybe simpler constraint can be used here: `if: :nic_provider_class_name`? All specs are green after this change.